### DISCUSSION TITLE
Season average rating overlays

### DISCUSSION THREAD

**Discussion by AlFontal** [meisnate12/Plex-Meta-Manager#1581] [Q&A]
Hi there! I am loving to play around with the different configs and I'm starting to get into a deeper level of customization of my setup.

Is there any way to get Season level ratings (and therefore, overlays) within the current defaults?

I might be able to write a script to do so, but I don't want to reinvent the wheel.

**Comment by chazlarson** [comment]
There is not, because none of the ratings sources provide ratings at a season level.

**Reply by AlFontal** [reply]
Yep, but computing the average rating of the episodes within a season would be trivial and provide a more-or-less accurate depiction of the season-level rating. 

As far as I know, Plex allows season level user ratings, I don't know if that's the case for critic/audience ones.
I am thinking of writing a script that is called as an operation and simply averages one of the rating scores of the episodes and updates the user season scores, which would then be used by a season-level overlay... 
Any pointers on how I might be able to use a custom script that is then called from the config files?

**Reply by chazlarson** [reply]
There is no provision for calling custom scripts.

**Reply by AlFontal** [reply]
Hmm... I see. However, if I run the script independently of PMM so that all my seasons are updated with user ratings of my choice, would I be able to use these lines:

```yml
    - pmm: ratings                                                    
      template_variables:
        rating1: user
        rating1_image: imdb
        overlay_level: season
``` 

So that the daily PMM run would pick up the user ratings pre-defined by the script (and stored in my Plex instance) and use it to generate overlays in the season posters? Or would the `overlay_level: season` cause it to simply not run?

**Reply by chazlarson** [reply]
That would cause an error since that's not a valid value in that context.  In the case of the default ratings overlay, the only valid "level" is "episode", as that particular overlay is only designed to work at the show/movie/episode level.  
<https://metamanager.wiki/en/nightly/defaults/overlays/ratings.html>

You can of course create whatever overlay you wish outside the default framework, but the default ratings overlay is not designed to do this.
<https://metamanager.wiki/en/nightly/metadata/overlay.html>

There is a special text variable for the user rating, which works at the season level:
<https://metamanager.wiki/en/nightly/metadata/overlay.html#special-text-variables>
